pkg/lib/_internal/dnf5daemon.ts

```diff
@@ -0,0 +1,324 @@
+        _dbus_client.addEventListener("close", () => {
+            console.warn("dnf5daemon went away from D-Bus");
+            _dbus_client = null;
+        });
+    }
+
+    return _dbus_client;
+}
+
+// Reconnect when privileges change
+superuser.addEventListener("changed", () => {
+    if (_dbus_client)
+        _dbus_client.close();
+    _dbus_client = null;
+});
+
+async function open_session(): Promise<string> {
+    const [session] = await call("/org/rpm/dnf/v0", "org.rpm.dnf.v0.SessionManager",
+                                 "open_session", [{}]) as string[];
+    return session;
+}
+
+function close_session(session: string) {
+    return call("/org/rpm/dnf/v0", "org.rpm.dnf.v0.SessionManager",
+                "close_session", [session]);
+}
+
+interface ListPackage {
+    arch: { t: "s", v: string }
+    download_size: { t: "t", "v": number }
+    id: { t: "i"; v: number };
+    is_installed: { t: "b"; v: boolean };
+    name: { t: "s"; v: string };
+    release: { t: "s"; v: string };
+    version: { t: "s"; v: string };
+}
+
+interface ResolvePackage {
+    arch: { t: "s"; v: string };
+    download_size: { t: "t"; v: number };
+    epoch: { t: "s"; v: string };
+    evr: { t: "s"; v: string };
+    from_repo_id: { t: "s"; v: string };
+    full_nevra: { t: "s"; v: string };
+    id: { t: "i"; v: number };
+    install_size: { t: "t"; v: number };
+    name: { t: "s"; v: string };
+    reason: { t: "s"; v: string };
+    release: { t: "s"; v: string };
+    repo_id: { t: "s"; v: string };
+    version: { t: "s"; v: string };
+}
+
+// TransactionItemType
+type object_type = "Package" | "Group" | "Environment" | "Module" | "Skipped";
+// TransactionItemAction
+type action = "Install" | "Upgrade" | "Downgrade" | "Reinstall" | "Remove" | "Replaced" | "Reset" | "Enable" | "Disable" | "Reason Change" | "Switch"
+// TransactionItemReason
+type reason = "User" | "Dependency" | "Clean" | "Group" | "None" | "Weak Dependency" | "External User"
+type TransactionItem = [
+    object_type,
+    action,
+    reason,
+    unknown,
+    ResolvePackage
+]
+
+type InstallResolveResult = [TransactionItem[], number]
+
+/**
+ * Call a dnf5daemon method
+ */
+function call(objectPath: string, iface: string, method: string, args?: unknown[], opts?: cockpit.DBusCallOptions) {
+    return dbus_client().call(objectPath, iface, method, args, opts);
+}
+
+export class Dnf5DaemonManager implements PackageManager {
+    name: string;
+
+    constructor() {
+        this.name = 'dnf5daemon';
+    }
+
+    async check_missing_packages(pkgnames: string[], progress_cb?: ProgressCB): Promise<MissingPackages> {
+        const data: MissingPackages = {
+            extra_names: [],
+            missing_ids: [],
+            missing_names: [],
+            unavailable_names: [],
+            remove_names: [],
+            download_size: 0,
+        };
+
+        if (pkgnames.length === 0)
+            return data;
```

Comment:
This added line is not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-22397-9ce44077-20250919-142346-fedora-42-devel/Coverage/lcov/github-pr.diff.gcov.html)